src/physics/em/detail/PhysicsConstants.hh

```diff
@@ -0,0 +1,46 @@
+//---------------------------------*-CUDA-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file PhysicsConstants
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/Types.hh"
+#include "base/Constants.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Physical constants which are derived from fundamental constants.
+ *
+ * Derived constants    | Unit                  | Notes
+ * -------------------- | --------------------- | ------------
+ * electron_mass_c2     | g * (m/s)^2           |
+ * electron_mass_mev    | g * (m/s)^2/MeV       | Geant4 electron mass
+ * migdal_constant      | cm^3                  | Bremsstrahlung
+ * lpm_constant         | MeV/cm                | Relativistic Bremsstrahlung
+ */
+
+using namespace constants;
+using namespace units;
+
+//!@{
+//! Derivative constants with units
+constexpr real_type electron_mass_c2  = electron_mass * c_light * c_light;
+constexpr real_type electron_mass_mev = electron_mass_c2
+                                        / (1e6 * e_electron * volt);
+constexpr real_type migdal_constant = 4 * pi * r_electron * lambdabar_electron
+                                      * lambdabar_electron;
+constexpr real_type lpm_constant = alpha_fine_structure * electron_mass_c2
+                                   * electron_mass_c2
+                                   / (2 * h_planck * c_light);
```

Comment:
Dropped electron_mass_mev as is is not used in the current code.  I think that defining this somewhere in physics will remove a lot of duplicated shared data (i.e., in various model data).

Reply:
Technically we could directly access the particle params, since there it's not duplicated, but it would result in a slowdown for the extra indirection. I think it's better also to use the mass from the particle data rather than a hardcoded constant to have greater consistency across the codebase. The only exception to this I'm aware of is where a measurement of the electron mass from CODATA 2018 is embedded into the classical electron radius, which is then used to derive some constants. To remove *that* inconsistency we could define a function to return the "constant" as a function of the electron mass, so we could store migdal's constant etc. as `__constant__` data rather than as compile-time constants... but that seems like a lot of effort to avoid a potential inconsistency in the 7th digit (or whatever) of our constants.